spec/support/assertions.rb

```diff
@@ -0,0 +1,19 @@
+module RSpec
+  module Support
+    module Assertions
+      def assert_successful(result, key = :foo)
+        expect(result).to be_success
+        expect(result.messages.fetch(key, [])).to be_empty
+      end
+
+      def refute_successful(result, messages, key = :foo)
+        expect(result).not_to be_success
+        expect(messages).to eq(result.messages.fetch(key))
+      end
```

Comment:
Just an idea. Since RSpec is more about expectations than assertions, maybe it would be better to call those methods `expect_successful` and `expect_not_successful`?